xc/xc7/fasm2bels/iob_models.py

```diff
@@ -465,9 +468,67 @@ def process_single_ended_iob(top, iob):
     if top_wire is not None:
         add_pull_bel(site, top_wire)
 
+    site.set_post_route_cleanup_function(cleanup_single_ended_iob)
     top.add_site(site)
 
 
+def cleanup_single_ended_iob(top, site):
+    """
+    Cleans up a single-ended IOB
+    - Detects whether there is an OBUFT with T input routed to const1. If so
+      then checks whether the ZINV_T1 inverter is enabled in the neighboring
+      OLOGIC site. If both cases are true then the OBUFT is replaced with an
+      OBUF.
+    """
+
+    # Check if there is an OBUFT
+    bel = site.maybe_get_bel("OBUFT")
+    if bel is None:
+        return
+
+    # Check if its T input is routed to const1
+    src = top.find_source_from_sink(site, "T")
+    if src != 1:
+        return
+
+    # Get the neighboring IOI3 tile prefix and loc
+    ioi_tile_loc = site.tile.rsplit("_", maxsplit=1)[1]
```

Comment:
Rather than doing a string replace here, why not walk 1 tile to the left or right?  That would be more robust?

Reply:
Fixed.